core/services/gateway/api/jsonrpc/services/vault/service.go

```diff
@@ -0,0 +1,259 @@
+
+	return &service{
+		methodConfig:      cfg,
+		donConfig:         donConfig,
+		don:               don,
+		lggr:              lggr.Named("VaultMethod"),
+		requestTimeoutSec: cfg.RequestTimeoutSec,
+		userRateLimiter:   userRateLimiter,
+		nodeRateLimiter:   nodeRateLimiter,
+		secretsStore:      make(map[string]map[string]SecretEntry),
+	}
+}
+
+func (s *service) Start(ctx context.Context) error {
+	return s.StartOnce("VaultMethod", func() error {
+		s.lggr.Info("starting vault method")
+		return nil
+	})
+}
+
+func (s *service) Close() error {
+	return s.StopOnce("VaultMethod", func() error {
+		s.lggr.Info("closing vault method")
+		return nil
+	})
+}
+
+func (s *service) HandleUserRequest(ctx context.Context, request *jsonrpc.Request, callbackCh chan<- *jsonrpc.Response) error {
+	s.lggr.Debugw("handling vault request", "method", request.Method, "id", request.ID)
+
+	// Create timeout context
+	timeoutCtx, cancel := context.WithTimeout(ctx, time.Duration(s.requestTimeoutSec)*time.Second)
+	defer cancel()
+
+	// Process request based on method
+	switch request.Method {
+	case MethodSecretsCreate:
+		return s.handleSecretsCreate(timeoutCtx, request, callbackCh)
+	default:
+		s.lggr.Debugw("unsupported method", "method", request.Method)
+		promHandlerError.WithLabelValues(s.donConfig.DonId, ErrUnsupportedMethod.Error()).Inc()
+
+		response := &jsonrpc.Response{
+			Version: "2.0",
+			ID:      request.ID,
+			Error: &jsonrpc.Error{
+				Code:    -32601, // Method not found
+				Message: "Method not found",
+				Data:    json.RawMessage(fmt.Sprintf("Unsupported method: %s", request.Method)),
+			},
+		}
+
+		select {
+		case callbackCh <- response:
+			return nil
+		case <-timeoutCtx.Done():
+			return timeoutCtx.Err()
+		}
+	}
+}
+
+func (s *service) handleSecretsCreate(ctx context.Context, request *jsonrpc.Request, callbackCh chan<- *jsonrpc.Response) error {
+	var req SecretsCreateRequest
+	if err := json.Unmarshal(request.Params, &req); err != nil {
+		response := &jsonrpc.Response{
+			Version: "2.0",
+			ID:      request.ID,
+			Error: &jsonrpc.Error{
+				Code:    -32602, // Invalid params
+				Message: "Invalid parameters",
+				Data:    json.RawMessage(fmt.Sprintf("Failed to parse request: %v", err)),
+			},
+		}
+		return s.sendResponse(ctx, response, callbackCh)
+	}
+
+	// Validate request
+	if req.ID == "" {
+		response := &jsonrpc.Response{
+			Version: "2.0",
+			ID:      request.ID,
+			Error: &jsonrpc.Error{
+				Code:    -32602, // Invalid params
+				Message: "Invalid parameters",
+				Data:    json.RawMessage("Secret ID cannot be empty"),
+			},
+		}
+		return s.sendResponse(ctx, response, callbackCh)
+	}
+
+	// Store secret
+	s.storeMu.Lock()
+	defer s.storeMu.Unlock()
+
+	// Extract sender from request metadata (this would come from JWT or other auth)
+	senderAddr := "default" // In a real implementation, extract from authenticated context
+	if s.secretsStore[senderAddr] == nil {
```

Comment:
This needs to happen in the Vault DON, not on a Gateway node.
The Gateway-DON message passing in the new format is missing.

Reply:
Maybe this code is only for demo, that's why it is not forwarding this call to a Vault Don currently?